plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/core/index/CorrelationParamsContext.java

```diff
@@ -0,0 +1,166 @@
+     * Parse into CorrelationParamsContext
+     * @param in Object
+     * @return CorrelationParamsContext
+     */
+    public static CorrelationParamsContext parse(Object in) {
+        if (!(in instanceof Map<?, ?>)) {
+            throw new MapperParsingException("Unable to parse CorrelationParamsContext");
+        }
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> contextMap = (Map<String, Object>) in;
+        VectorSimilarityFunction similarityFunction = VectorSimilarityFunction.EUCLIDEAN;
+        Map<String, Object> parameters = new HashMap<>();
+
+        if (contextMap.containsKey(VECTOR_SIMILARITY_FUNCTION)) {
+            Object value = contextMap.get(VECTOR_SIMILARITY_FUNCTION);
+
+            if (value != null && !(value instanceof String)) {
+                throw new MapperParsingException(String.format(Locale.getDefault(), "%s must be a string", VECTOR_SIMILARITY_FUNCTION));
+            }
+
+            try {
+                similarityFunction = VectorSimilarityFunction.valueOf((String) value);
+            } catch (IllegalArgumentException ex) {
+                throw new MapperParsingException(String.format(Locale.getDefault(), "Invalid %s: %s", VECTOR_SIMILARITY_FUNCTION, value));
+            }
+        }
+        if (contextMap.containsKey(PARAMETERS)) {
+            Object value = contextMap.get(PARAMETERS);
+            if (value == null) {
+                parameters = null;
+            }
+
+            if (!(value instanceof Map)) {
+                throw new MapperParsingException("Unable to parse parameters for Correlation context");
+            }
+
+            @SuppressWarnings("unchecked")
+            Map<String, Object> valueMap = (Map<String, Object>) value;
+            parameters.putAll(valueMap);
+        }
+        return new CorrelationParamsContext(similarityFunction, parameters);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(VECTOR_SIMILARITY_FUNCTION, similarityFunction.name());
+        if (params == null) {
+            builder.field(PARAMETERS, (String) null);
+        } else {
+            builder.startObject(PARAMETERS);
+            for (Map.Entry<String, Object> parameter : parameters.entrySet()) {
+                builder.field(parameter.getKey(), parameter.getValue());
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CorrelationParamsContext that = (CorrelationParamsContext) o;
+        return similarityFunction == that.similarityFunction && parameters.equals(that.parameters);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(similarityFunction, parameters);
+    }
```

Comment:
Do you need equals and hashcode? The `parameters` map has values of type `Object` so you have no way of knowing whether those types will use identity based equals and hash code, which can lead to seemingly identical instances not being equal. I would recommend to either remove these and use identity-based equality for this class, or else constrain the value type of `parameters` to a type that implements proper equals/hash code.

Reply:
removed `equals` & `hashCode` method.